doc/auto_techsupport_and_coredump_mgmt.md

```diff
@@ -68,14 +76,52 @@ The naming format and compression is governed by the script `/usr/local/bin/core
 
 Where `<comm>` value in the command name associated with a process. comm value of a running process can be read from `/proc/[pid]/comm` file
 
-## 4. Schema Additions
+## 4. Memory leak based techsupport invokation
+
+If the following condition resolves to true:
+```
+mem_usage > mem_usage_threshold || ${container}_mem_usage > ${container}_mem_usage_threshold
+```
+
+where ```mem_usage``` is total system memory used (MemAvailable from /proc/meminfo),
+```mem_usage_threshold``` configured threashold,
+```${container}_mem_usage``` used memory by $container ("docker stats --no-stream --format {{.MemUsage}}" $container),
+```${container}_mem_usage_threshold``` configured memory threshold for $contianer, 
+
+the SONiC techsupport is automatically generated.
+
+The check will be implemented as a script that is ran by monit periodically:
+
+```
+check program mem_checker with path "/usr/bin/mem_check"
+    if status != 0 for 10 times within 20 cycles then exec /usr/local/bin/coredumpgen_handler"
+```
+
+The action is going to be ran only once the mem_check script detects memory usage above threshold.
+
+The "10 times within 20 cycles" part is kept in sync with mem_usage alert from sonic-host monit configuration.
+It is possible to make those values configurable however, only the threshold value is considered to be configurable.
+
+The rate limit as well as techsupport maximum limit is applicable to techsupports generated by memory check.
+
+#### 202106 and older
+
+To support thechsupport generation on memory leaks a simple rule to monit is added:
+
+```
+check system $HOST
+    if memory usage > 90% for 10 times within 20 cycles then exec /usr/bin/generate_dump
+```
+
+## 5. Schema Additions
 
 ### Config DB
 
 #### AUTO_TECHSUPPORT Table
 ```
 key                    = "AUTO_TECHSUPPORT|global"  
 state                  = "enabled" / "disabled"    ; Enable this to make the Techsupport Invocation event driven based on core-dump generation
+mem_threshold          = 1*2DIGIT                  ; Memory threashold; 0 to disable techsupport invokation on mem leak.

```

Comment:
@stepanblyschak  as this is for enable/disable the for memory i dont think the name threshold is correct or reflecting the state. can you please think of a better name? this will also affect the cli name